cross/python310/Makefile

```diff
@@ -25,7 +25,29 @@ CONFIGURE_ARGS += --enable-ipv6
 CONFIGURE_ARGS += --without-ensurepip
 CONFIGURE_ARGS += --enable-loadable-sqlite-extensions
 CONFIGURE_ARGS += --with-computed-gotos=yes
+
+include ../../mk/spksrc.archs.mk
+
+# optionally generate optimized code
+ifeq ($(strip $(PYTHON_OPTIMIZE)),1)
+
+# old compilers fail with: unrecognized command line option "-fprofile-correction"
+ifneq ($(findstring $(ARCH),$(OLD_PPC_ARCHS)),$(ARCH))
 CONFIGURE_ARGS += --enable-optimizations
+endif
+
+# old compilers fail with unrecognized command line options: -flto -fuse-linker-plugin -ffat-lto-objects -flto-partition=none
+ifneq ($(findstring $(ARCH),$(ARMv5_ARCHS) $(OLD_PPC_ARCHS)),$(ARCH))
+# Enable Link-Time Optimization
+CONFIGURE_ARGS += --with-lto
+endif
+
+endif
+
+ifeq ($(findstring $(ARCH),$(OLD_PPC_ARCHS)),$(ARCH))
+# older gcc does not know -Wno-unused-result
+CONFIGURE_ARGS += ac_cv_disable_unused_result_warning=no
+endif
```

Comment:
I'm ok with the code but I thought we would get rid of DSM5 support as this is rather costly...

Reply:
I know, but I have a DS-210+ that I personally want to keep alive as long as possibel...